packages/core/src/browser/frontend-application.ts

```diff
@@ -41,6 +42,13 @@ export interface FrontendApplicationContribution {
      */
     onStart?(app: FrontendApplication): MaybePromise<void>;
 
+    /**
+     * If this function returns something evaluating to true, we will try to prevent the window unloading,
+     * and eventually display the message returned if you return a string.
+     * (message display depends on the browser, some don't do it)
+     */
+    preventStop?(app: FrontendApplication): string | number | boolean | undefined | void;
```

Comment:
`onWillStop?(app): any`?

We use `Will` for event naming to point out whether something is going to hapen.

Reply:
I'm not strongly against it, but if we allow extensions to manage their own display, and just prevent or not based on the return, wouldn't `preventStop?(app): boolean` better fit?

The goal of this API is for people to prevent the closing, not really to notify you that it is going to close. Wdyt?

Reply:
Ok, I went with `onWillStop`.